include/pybind11/common.h

```diff
@@ -608,13 +608,24 @@ template <typename T> struct is_fmt_numeric<T, enable_if_t<std::is_arithmetic<T>
 NAMESPACE_END(detail)
 
 template <typename T> struct format_descriptor<T, detail::enable_if_t<detail::is_fmt_numeric<T>::value>> {
-    static constexpr const char c = "?bBhHiIqQfdgFDG"[detail::is_fmt_numeric<T>::index];
-    static constexpr const char value[2] = { c, '\0' };
-    static std::string format() { return std::string(1, c); }
+    static constexpr const char c1 = "?bBhHiIqQfdgZZZ"[detail::is_fmt_numeric<T>::index];
+    static constexpr const char c2 = "\0\0\0\0\0\0\0\0\0\0\0\0fdg"[detail::is_fmt_numeric<T>::index];
```

Comment:
I have to say, I'm not a big fan of this part.  I think it would be cleaner to just remove `FDG` from here and specialize the whole `format_descriptor` struct for `std::complex` inside `complex.h`.